docs: 2.3.0 changelog (#139)
canonical/craft-grammar: docs/changelog.rst

.. _release 2.3.0:

> 2.3.0 (2025-09-17)

Author: ```suggestion
2.3.0 (2025-09-19)
```

------------------

New features:

- The ``checker`` option for ``GrammarProcessor`` is now optional.

> - Add ``any`` as an always-available platform for ``for`` grammar that is always
>   true.

Reviewer: ```suggestion
- Add the ``any`` platform.  ``for`` statements that select against this platform
  are always true.
```

> - Add ``valid_platforms`` and ``valid_architectures`` optional arguments to
>   ``GrammarProcessor`` to allow validation of platforms and architectures listed in
>   the grammar.

Reviewer: ```suggestion
- Add the optional ``valid_platforms`` and ``valid_architectures`` arguments to
  ``GrammarProcessor``, which allow for the validation of platforms and architectures
  listed in the grammar.
```

> - Add a legal ``else`` clause to ``for`` grammar.

Reviewer: ```suggestion
- Add an ``else`` clause to the ``for`` statement.
```
Have we ever added an _illegal_ clause?

> - An app using ``GrammarProcessor`` can choose a specific grammar variant.

Reviewer: ```suggestion
- Apps that use ``GrammarProcessor`` can now choose a specific grammar variant.
```
